### PR TITLE
Implement logger and add colum 'is nullable'

### PR DESCRIPTION
**Pull request by Thaneshwor**
1. Change column header 'Actual Datatype' to 'Maximum Datatype' while displaying columns and their maximum datatype available.
2. Add column 'is nullable' while displaying columns and available data types and remove 'None' from column 'datatypes' .
3. Add code for generating error.log file 
4.Code change for finding datatype
5. Logic change for finding an erroneous column
5. Code refactor

### CODE REVIEW

File: data_profiller/utils/data_collection.py [hunk 1]
```diff
@@ -93,25 +93,46 @@ def get_init_dictionary(headings):
...
+
+def isInt(val):
+    '''
+    Return true if input parameter is of type integer.
+    '''
+    if '.' in str(val):
+        return False
+
+    try:
+
```
Comment (romitamgai) [inline data_profiller/utils/data_collection.py]: no need for extra line

File: data_profiller/utils/data_collection.py [hunk 2]
```diff
@@ -93,25 +93,46 @@ def get_init_dictionary(headings):
     return column_datatype_line_no
 
 
+def isFloat(val):
+    '''
+    Return true if input parameter is of type float.
+    '''
+    try:
+        x = float(val)
+        return True
```
Comment (romitamgai) [inline data_profiller/utils/data_collection.py]: add an extra line before return

File: data_profiller/utils/data_collection.py [hunk 3]
```diff
@@ -93,25 +93,46 @@ def get_init_dictionary(headings):
     return column_datatype_line_no
 
 
+def isFloat(val):
+    '''
+    Return true if input parameter is of type float.
+    '''
+    try:
+        x = float(val)
```
Comment (romitamgai) [inline data_profiller/utils/data_collection.py]: no need for variable assignment here

File: data_profiller/utils/data_collection.py [hunk 4]
```diff
@@ -93,25 +93,46 @@ def get_init_dictionary(headings):
...
+    '''
+    Return true if input parameter is of type integer.
+    '''
+    if '.' in str(val):
+        return False
+
+    try:
+
+        return isinstance(int(float(val)), int)
+
```
Comment (romitamgai) [inline data_profiller/utils/data_collection.py]: no need for extra line

File: data_profiller/utils/file_operation.py [hunk 5]
```diff
@@ -20,16 +24,20 @@ def get_dataframe(file_name):
...
+            if header.find(";") != -1:
+                return ";"
+            if header.find(",") != -1:
+                return ","
+            if header.find("\t") != -1:
+                return "\t"
+            if header.find("|") != -1:
+                return "|"
+    except Exception as e:
+        print('error occuure---------------------', e)
```
Comment (romitamgai) [inline data_profiller/utils/file_operation.py]: typo

**Review by romitamgai — CHANGES_REQUESTED** [review]

File: data_profiller/utils/data_collection.py [hunk 6]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
+
 def get_datatype(cell_value):
-    ''' Return datatype of input parameter '''
+    '''Return datatype of input parameter in string formate'''
 
     if cell_value == 'NULL':
         dtype_cell = None
+    elif is_date(cell_value):
+        dtype_cell = 'date'
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]: Why are you assigning these values if all you do is return them later? You can return right away.
```suggestion
        return 'date'
```

File: data_profiller/utils/data_collection.py [hunk 7]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
+    elif isInt(cell_value):
+        dtype_cell = 'integer'
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]:
```suggestion
        return 'integer'
```

File: data_profiller/utils/data_collection.py [hunk 8]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
+    elif isFloat(cell_value):
+        dtype_cell = 'float'
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]:
```suggestion
        return 'float'
```

File: data_profiller/utils/data_collection.py [hunk 9]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
+
+        return False
+
+
 def get_datatype(cell_value):
-    ''' Return datatype of input parameter '''
+    '''Return datatype of input parameter in string formate'''
 
     if cell_value == 'NULL':
         dtype_cell = None
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]:
```suggestion
        return None
```

File: data_profiller/utils/data_collection.py [hunk 10]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
+    elif is_date(cell_value):
+        dtype_cell = 'date'
+    elif isInt(cell_value):
+        dtype_cell = 'integer'
+    elif isFloat(cell_value):
+        dtype_cell = 'float'
     else:
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py:109]:
```suggestion

```

File: data_profiller/utils/data_collection.py [hunk 11]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
-                if '.' in cell_value:
-                    dtype_cell = 'float'
-
-                if int(cell_value):
-                    dtype_cell = 'integer'
+        dtype_cell = 'string'
 
-        except:
-            pass
     return dtype_cell
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]:
```suggestion

```

**Review by SafalPandey — CHANGES_REQUESTED** [review]
Use early returns.

File: data_profiller/utils/data_collection.py [hunk 12]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
-            if is_date(cell_value):
-                dtype_cell = 'date'
-            else:
-                dtype_cell = 'string'
-                if '.' in cell_value:
-                    dtype_cell = 'float'
-
-                if int(cell_value):
-                    dtype_cell = 'integer'
+        dtype_cell = 'string'
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]:
```suggestion
    return 'string'
```

File: data_profiller/utils/data_collection.py [hunk 13]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
...
+    try:
+        float(val)
+
+        return True
+    except:
+
+        return False
+
+
+def isInt(val):
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]: Use snake case.
```suggestion
def is_int(val):
```

File: data_profiller/utils/data_collection.py [hunk 14]
```diff
@@ -93,31 +93,57 @@ def get_init_dictionary(headings):
     return column_datatype_line_no
 
 
+def isFloat(val):
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]:
```suggestion
def is_float(val):
```

File: data_profiller/utils/data_collection.py [hunk 15]
```diff
@@ -93,31 +93,55 @@ def get_init_dictionary(headings):
...
+    '''
+    Return true if input parameter is of type integer.
+    '''
+    if '.' in str(val):
+
+        return False
+
+    try:
+
+        return isinstance(int(float(val)), int)
```
Comment (SafalPandey) [inline data_profiller/utils/data_collection.py]: What's the point of `float(val)` here? Add a comment.

**Review by romitamgai — APPROVED** [review]

**Pull request merged**